astropy/uncertainty/function_helpers.py

```diff
@@ -62,6 +71,20 @@
 dispatched_function = FunctionAssigner(DISPATCHED_FUNCTIONS)
 
 
+def is_distribution(x):
+    from astropy.uncertainty import Distribution
```

Comment:
Yes, sadly. I guess the alternative would be to import `FUNCTION_HELPERS`, etc., inside `Distribution.__array_function__`, but then every function call is slowed down.

Reply:
Or we define a Protocol / ABC in a third file. No circular imports or slow-downs.

Reply:
Any opinion on adding a `DistributionBase` or `AbstractDistribution` Protocol? e.g. in a new file `base.py`:

```python
from typing import runtime_checkable, Protocol

class AbstractDistribution(Protocol):
    @property
    def distribution(self): ...

    @property
    def dtype(self): ...

    @property
    def n_samples(self): ...
```

Reply:
Then we can check for `Distribution`-like stuff in an implementation-free manner and avoid circular imports.

Reply:
Something like that may be good, though I'd prefer to first revisit your old PR of merging how `Distribution` and `Masked` generate classes, etc. While doing the work on `Distribution`, I realized there are quite a few bits that are not all that logical, and are done better for `Masked`. But of course the problem is to find the time...

Reply:
Yeah, the issue I had with this abstract class generation technique was making it so that the wrapper classes could stack in an intuitive and introspectable way so that we could have a clean MRO and way to unwrap everything.

Reply:
So defer this to a future PR?

Reply:
Yes, let's. It would be nice to solve this for all classes that use `__array_function__` (`Distribution`, `Masked`, `Quantity`).